testplan/common/utils/remote.py

```diff
@@ -135,3 +137,28 @@ def rm_cmd(path):
 def filepath_exist_cmd(path):
     """Checks if filepath exists."""
     return " ".join(["test", "-e", path])
+
+
+def calculate_pool_size_by_tasks(
+    tasks: Iterable[Task], plan_runtime_target: int
+):
```

Comment:
shall we do tasks None/empty check upfront? btw, what is our behavior if discover no task and we call schedule with None/empty list? Might worth checking. e.g do we raise expectation, do we exit 0 or 1?